### PR TITLE
feat: update view in room, allow more instant placement

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [MOPLAT-654] 

### Description

Adds a new view in room experience for devices with lidar support. 

https://user-images.githubusercontent.com/49686530/233207243-f541b08e-a6fd-4496-989b-c90a90b09aa4.mov

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

- More instant view in room placement - brian 

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[MOPLAT-654]: https://artsyproduct.atlassian.net/browse/MOPLAT-654?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ios/Artsy/Resources/Artsy-Bridging-Header.h [hunk 1]
```diff
@@ -116,3 +116,10 @@
 
 // Sentry Swift Access
 #import "ARSentryReporter.h"
+
+// AR
+#import "ARAugmentedRealityConfig.h"
+#import "ARAugmentedVIRSetupViewController.h"
+#import "ARInformationView.h"
+#import "ARSCNWallNode.h"
+#import "SCNArtworkNode.h"
```
Comment (brainbicycle) [inline ios/Artsy/Resources/Artsy-Bridging-Header.h:125]: swift access for some of the old AR classes

File: ios/Artsy/View_Controllers/ARVIR/Nodes/AppleARCode/FocusSquareSegment.swift [hunk 2]
```diff
@@ -0,0 +1,141 @@
+/*
+See LICENSE folder for this sample’s licensing information.
+
+Abstract:
+Corner segments for the focus square UI.
+*/
+
+import SceneKit
+
+extension FocusSquare {
```
Comment (damassi) [inline ios/Artsy/View_Controllers/ARVIR/Nodes/AppleARCode/FocusSquareSegment.swift:10]: Did we need to copy / augment 3rd party code here?

Comment (brainbicycle) [inline ios/Artsy/View_Controllers/ARVIR/Nodes/AppleARCode/FocusSquareSegment.swift:10]: this and all the code in the AppleARCode directory is code from Apple's example project here: https://developer.apple.com/documentation/arkit/environmental_analysis/placing_objects_and_handling_3d_interaction
It is MIT licensed and the license is kept in the folder. 
Mostly it is the code for the focusSquare (3d cursor) that does nice things like animate open and closed, pulse and adapt to planes.

**Review by damassi — DISMISSED** [review]
One question around some apple code, but otherwise LGTM!

File: ios/Artsy/View_Controllers/ARVIR/Nodes/AppleARCode/FocusSquareSegment.swift (hunk 2 above)
Comment (damassi) [reply ios/Artsy/View_Controllers/ARVIR/Nodes/AppleARCode/FocusSquareSegment.swift:10]: Haha i think we just jinxed ourselves. Cool, ty for explaining 👍

File: ios/Artsy/View_Controllers/ARVIR/Nodes/VirtualArtwork.swift [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import Foundation
+import SceneKit
+import ARKit
+
+class VirtualArtwork: SCNNode {
```
Comment (brainbicycle) [inline ios/Artsy/View_Controllers/ARVIR/Nodes/VirtualArtwork.swift:5]: this class represents the virtual artwork node placed in the users space, it is initialized with the config that contains the artworks dimensions and image info which is converted to the 3d artwork. Attaching the tracked raycast queries to the object allows updating the objects placement as ARKit becomes more aware of the surroundings.

File: ios/Artsy/View_Controllers/ARVIR/VCs/ARAugmentedVIRSetupViewController.m [hunk 4]
```diff
@@ -235,17 +239,31 @@ - (void)next
...
         } else {
             [self requestDenied];
         }
     }];
 }
 
+- (void)showVIRVC
+{
+    BOOL hasLidarEnabledDevice = [ARAugmentedVIRSetupViewController hasLidarEnabledDevice];
+    if (hasLidarEnabledDevice && _enableInstantVIR) { // Lidar device we can do instant vertical detection
```
Comment (brainbicycle) [inline ios/Artsy/View_Controllers/ARVIR/VCs/ARAugmentedVIRSetupViewController.m:252]: this old AR setup vc will get called on the first launch of VIR, it shows an onboarding video and asks the user to enable camera access. Only real change is checking the feature flag and if lidar is enabled here as well and if so showing the new experience.

File: src/app/Scenes/Artwork/Components/ArtworkActions.tsx [hunk 5]
```diff
@@ -43,6 +43,8 @@ export const ArtworkActions: React.FC<ArtworkActionsProps> = ({ artwork, shareOn
   const { trackEvent } = useTracking()
   const space = useSpace()
 
+  const enableInstantVIR = useFeatureFlag("AREnableInstantViewInRoom")
```
Comment (brainbicycle) [inline src/app/Scenes/Artwork/Components/ArtworkActions.tsx:46]: pass feature flag from typescript side so we can disable if needed

File: src/app/utils/DevMenu.tsx [hunk 6]
```diff
@@ -250,15 +247,15 @@ export const DevMenu = ({ onClose = () => dismissModal() }: { onClose(): void })
               Sentry.nativeCrash()
             }}
           />
-          <FeatureFlagMenuItem title={`Sentry release name: "${eigenSentryReleaseName()}"`} />
-          <FeatureFlagMenuItem
+          <DevMenuButtonItem title={`Sentry release name: "${eigenSentryReleaseName()}"`} />
+          <DevMenuButtonItem
```
Comment (brainbicycle) [inline src/app/utils/DevMenu.tsx:251]: noise, apologies but not all these items are feature flags so the name doesn't really work.

File: ios/Artsy/View_Controllers/Live_Auctions/LiveActionStateReconciler.swift [hunk 7]
```diff
@@ -90,6 +90,9 @@ extension PublicFunctions: LiveAuctionStateReconcilerType {
     }
 
     func processCurrentLotUpdate(_ update: AnyObject) {
+        guard let update = update as? [String: AnyObject] else {
```
Comment (brainbicycle) [inline ios/Artsy/View_Controllers/Live_Auctions/LiveActionStateReconciler.swift:93]: noise: oddly started getting a compilation error after adding some more files to the bridging header 😨 this fixed it

File: ios/Artsy/View_Controllers/ARVIR/VCs/ARAugmentedWallBasedVIRViewController.swift [hunk 8]
```diff
@@ -0,0 +1,437 @@
+import UIKit
+import ARKit
+
+class ARAugmentedWallBasedVIRViewController: UIViewController {
```
Comment (brainbicycle) [inline ios/Artsy/View_Controllers/ARVIR/VCs/ARAugmentedWallBasedVIRViewController.swift:4]: this class is the meat of the feature

File: ios/Artsy/View_Controllers/ARVIR/VCs/ARAugmentedWallBasedVIRViewController.swift [hunk 9]
```diff
@@ -0,0 +1,437 @@
...
+    }
+}
+
+
+/// - Tag: CoachingOverlayViewDelegate
+extension ARAugmentedWallBasedVIRViewController: ARCoachingOverlayViewDelegate {
+
+    /// - Tag: HideUI
+    func coachingOverlayViewWillActivate(_ coachingOverlayView: ARCoachingOverlayView) {
+        // upperControlsView.isHidden = true
```
Comment (brainbicycle) [inline ios/Artsy/View_Controllers/ARVIR/VCs/ARAugmentedWallBasedVIRViewController.swift:384]: TODO: probably should be hiding reset button and other UI here

File: ios/Artsy/View_Controllers/ARVIR/Views/ARInformationView.h [hunk 10]
```diff
@@ -5,6 +5,7 @@
 @property (nonatomic, copy, nonnull) NSString *xOutOfYMessage;
 @property (nonatomic, copy, nonnull) NSString *bodyString;
 @property (nonatomic, strong, nonnull) UIView *contents;
+@property (nonatomic, copy, nullable) NSString *stateTag;
```
Comment (brainbicycle) [inline ios/Artsy/View_Controllers/ARVIR/Views/ARInformationView.h:8]: the old information view that shows up at bottom of screen at each stage of the AR experience, added this stateTag to easily tell which stage the view is currently at

File: src/app/store/config/features.ts [hunk 11]
```diff
@@ -242,6 +242,11 @@ export const features = defineFeatures({
     showInDevMenu: true,
     echoFlagKey: "AREnableSWALandingPageTestimonials",
   },
+  AREnableInstantViewInRoom: {
+    description: "Enable Instant View In Room",
+    readyForRelease: false,
+    showInDevMenu: true,
+  },
```
Comment (gkartalis) [inline src/app/store/config/features.ts:249]: I guess we will mark it as readyForRelease in a different PR (not sure what's the timeline on the release of this 👀)

File: ios/Podfile [hunk 12]
```diff
@@ -160,6 +160,7 @@ target 'Artsy' do
 
     installer.pods_project.targets.each do |target|
         target.build_configurations.each do |config|
+            config.build_settings["DEVELOPMENT_TEAM"] = "23KMWZ572J"
```
[the remark this answers is not in the document]
Comment (brainbicycle) [reply ios/Podfile:163]: 👍 I came across signing issues in pods on Xcode 14.3 but I will remove for now and bring it back if others experience

[the remark this answers is not in the document]
Comment (gkartalis) [reply ios/Podfile:163]: makes sense!

File: src/app/store/config/features.ts (hunk 11 above)
Comment (brainbicycle) [reply src/app/store/config/features.ts:249]: yeah I was thinking people could mess with it behind the flag a bit first and give feedback if anything needs changing

**Pull request merged**